src/loaders/OMETiffLoader.js

```diff
@@ -360,4 +352,46 @@ export default class OMETiffLoader {
       tiff.ifdRequests[index] = tiff.parseFileDirectoryAt(offsets[index]);
     }
   }
+
+  /**
+   * Trunactes tiles on the edge to match the height/width reported by the image.
+   * @param {tileData: TypedArray[]} data The array to be filled in.
+   * @param {Object} tile { x, y, z }
+   * @returns {TypedArray} TypedArray
+   */
+  _truncateTiles(data, tile) {
+    const { x, y, z } = tile;
+    const { tileSize } = this;
+    let height = tileSize;
+    let width = tileSize;
+    const {
+      height: zoomLevelHeight,
+      width: zoomLevelWidth
+    } = this.getRasterSize({ z });
+    const maxXTileCoord = Math.floor(zoomLevelWidth / tileSize);
+    const maxYTileCoord = Math.floor(zoomLevelHeight / tileSize);
+    if (x === maxXTileCoord) {
+      width = zoomLevelWidth % tileSize;
+    }
+    if (y === maxYTileCoord) {
+      height = zoomLevelHeight % tileSize;
+    }
+    if (
+      (width === tileSize && height === tileSize) ||
+      data[0].length === width * height
```

Comment:
why will we return early if `data[0].length === width * height`, regardless of (`width === tileSize && height === tileSize`)? It feels like we want to truncate always if `width` or `height` are not `tilesize`?

Reply:
I guess my question is more, what's the purpose of `data[0].length === width * height`?

Reply:
This prevented the zarr stores, for instance, from having a truncation operation (although that is no longer valid).  More generally, though, the lower levels of the image pyramid for Bioformats2raw are actually not properly padded so they are the "true" size of the image at that resolution.  This is helpful because it lets us transfer less data to get the lower levels of the pyramid (like how the `GridLayer` in vizarr uses a very tiny image which allows it to load more quickly).